packages/contracts-bedrock/src/deploy-utils.ts

```diff
@@ -314,20 +316,24 @@ export const isStep = async (
 /**
  * Mini helper for executing a given step.
  *
+ * @param hre HardhatRuntimeEnvironment.
  * @param opts Options for executing the step.
  * @param opts.isLiveDeployer True if the deployer is live.
  * @param opts.SystemDictator SystemDictator contract.
  * @param opts.step Step to execute.
  * @param opts.message Message to print before executing the step.
  * @param opts.checks Checks to perform after executing the step.
  */
-export const doStep = async (opts: {
-  isLiveDeployer?: boolean
-  SystemDictator: ethers.Contract
-  step: number
-  message: string
-  checks: () => Promise<void>
-}): Promise<void> => {
+export const doStep = async (
+  hre: HardhatRuntimeEnvironment,
+  opts: {
+    isLiveDeployer?: boolean
+    SystemDictator: ethers.Contract
+    step: number
+    message: string
+    checks: () => Promise<void>
+  }
+): Promise<void> => {
   if (!(await isStep(opts.SystemDictator, opts.step))) {
```

Comment:
You have a provider with the system dictator contract, can use that to query the chainid instead of passing thru the hre through multiple layers in many places